internal/context/context.go

```diff
@@ -135,24 +135,33 @@ func makeDir(filePath string) error {
 	return nil
 }
 
-func SignNFCert(nfType, nfId string) error {
-	nfCertPath := openapi.GetNFCertPath(factory.NrfConfig.GetCertBasePath(), nfType)
+func SignNFCert(nfType, nfId, nfCertPem string) error {
+	// Use default {Nf_type}.pem
+	nfCertPath := oauth.GetNFCertPath(factory.NrfConfig.GetCertBasePath(), nfType, "")
+	newCertPath := oauth.GetNFCertPath(factory.NrfConfig.GetCertBasePath(), nfType, nfId)
+
+	logger.NfmLog.Infoln("Use NF certPath:", nfCertPath)
 
 	// Get NF's Certificate from file
-	nfCert, err := openapi.ParseCertFromPEM(nfCertPath)
+	nfCert, err := oauth.ParseCertFromPEM(nfCertPath)
 	if err != nil {
 		logger.NfmLog.Warnf("No NF cert: %v; generate new one", err)
 
+		_, err = oauth.GenerateRSAKeyPair(nfCertPath, "")
+		if err != nil {
+			return errors.Wrapf(err, "Generate Error")
+		}
+
 		// Get NF's Public key from file
 		var nfPubKey *rsa.PublicKey
-		nfPubKey, err = openapi.ParsePublicKeyFromPEM(nfCertPath)
+		nfPubKey, err = oauth.ParsePublicKeyFromPEM(nfCertPath)
 		if err != nil {
 			return errors.Wrapf(err, "sign NF cert")
```

Comment:
line 150 `oauth.GenerateRSAKeyPair(nfCertPath, "")` should be moved here (when ParsePublicKayFromPEM failed)